docs/security.md

```diff
@@ -96,13 +96,150 @@ The Dockerfile is configured to run as a non-root user:
+AUDIT_ENABLED=false
+
+# File-only audit logging (no database storage)
+AUDIT_ENABLED=true
+AUDIT_LOG_FILE=./data/audit.log
+AUDIT_STORE_IN_DB=false
+
+# Database-only audit logging (no file)
+AUDIT_ENABLED=true
+AUDIT_LOG_FILE=""
+AUDIT_STORE_IN_DB=true
+```
+
+#### Privacy and Data Protection
+
+**Token Obfuscation Guarantees:**
+- All tokens in audit logs are automatically obfuscated using a consistent pattern
+- Original tokens are never stored in audit logs in plaintext
+- Obfuscation preserves prefix (e.g., `tok-`) and shows partial content for identification
+- For tokens with known prefixes (such as `tok-`, `sk-`, `api_`, `Bearer `, or `ghp_`): the obfuscation logic shows the first 4 and last 4 characters of the token, with asterisks in the middle.
+- Example: `tok-1234567890abcdef` becomes `tok-1234****cdef`
+- For tokens without a recognized prefix, the obfuscation may show only the first and last 2 characters, or fully mask the value depending on configuration.
+- Example: `abcd1234efgh5678` (no known prefix) becomes `ab********78`
+
+**No-Secrets Policy:**
+- API keys are never logged in audit events
+- Request/response bodies containing sensitive data are not logged
+- Only metadata (method, endpoint, status, duration) is captured
+- Personal Identifiable Information (PII) is not logged
+
+**Data Minimization:**
+- Audit events contain only necessary information for security investigations
+- Client IP addresses are logged but can be masked via configuration
+- User-Agent strings are logged for security analysis
+- Request IDs enable correlation without exposing sensitive data
+
+#### Retention Guidance
+
+**File Backend Retention:**
+- Implement log rotation using standard tools (logrotate, Docker volume rotation)
+- Recommended retention: 90 days minimum for compliance, 1 year for enhanced security
+- Secure storage with appropriate file permissions (600 or 640)
+- Consider encryption at rest for sensitive environments
+
+```bash
+# Example logrotate configuration
+./data/audit.log {
+    daily
+    rotate 90
+    compress
+    delaycompress
+    missingok
+    notifempty
+    copytruncate
+}
+```
+
+**Database Backend Retention:**
+- Database storage enables structured queries and analytics
+- Implement automated cleanup of old audit events
+- Consider partitioning by date for large deployments
+- Recommended indexes on timestamp, action, and project_id fields
+
+```sql
+-- Example cleanup query (run via scheduled job)
+-- For databases that support LIMIT in DELETE (e.g., MySQL, SQLite):
+DELETE FROM audit_events 
+WHERE timestamp < datetime('now', '-90 days')
+LIMIT 1000;
+
+-- Repeat the cleanup job until no more rows match the condition.
+
+-- For databases that do not support LIMIT in DELETE (e.g., PostgreSQL), consider using a CTE:
+-- WITH del AS (
+--   SELECT ctid FROM audit_events WHERE timestamp < NOW() - INTERVAL '90 days' LIMIT 1000
+-- )
+-- DELETE FROM audit_events WHERE ctid IN (SELECT ctid FROM del);
+
```

Comment:
Added DELETE ... LIMIT example and PostgreSQL CTE variant for batched retention cleanup.